src/store/SwapMetaView.cc

```diff
@@ -84,7 +84,7 @@ Store::SwapMetaView::checkExpectedLength(const size_t expectedLength) const
 {
     if (rawLength != expectedLength)
         throw TextException(ToSBuf("Bad value length in a Store entry meta field expecting a ",
-                                   expectedLength, "-byte value: ", *this), Here());
+                                   expectedLength, "-byte value: ", this), Here());
```

Comment:
```
../../../src/store/SwapMetaView.cc:86:35:   required from here
../../../src/sbuf/Stream.h:66:20: error: no match for 'operator<<' (operand types are 'std::basic_ostream<char>' and 'const Store::SwapMetaView')
   66 |     (out << ... << args);
      |     ~~~~~~~~~~~~~~~^~~~~
```

Reply:
The error message quoted in this change request is about "printing" SwapMetaView. We do have a "printing" operator for SwapMetaView:

```C++
/// writes a short human-readable summary of the given SwapMetaView object
std::ostream &operator <<(std::ostream &, const Store::SwapMetaView &);
```

It is the operator that should be (and, I presume, was) used in this code before C++17 migration AFAICT. I do not know what the correct bug fix is because I do not understand exactly why GCC v12 stopped "seeing" the existing "printing" operator declared for SwapMetaView in this context.

@yadij, does GCC report any "candidates" for this compilation failure? If it does, is there a SwapMetaView-related candidate? Those candidate messages often hint at the true cause of the problem. If you can share them (here or, if there are many candidates, in a GitHub gist), please do.

----

If we cannot find the right fix quickly enough, we can undo the corresponding code-readability improvement (see commit https://github.com/squid-cache/squid/pull/1212/commits/6730aa18c38091be26755017c2e7e25984bdc70c).

Reply:
I have placed a copy of "make -k" build log at http://treenet.co.nz/projects/squid/gcc12.log